publish-packages.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "Publishes the NuGet packages to the specified location."
+    echo "For publishing to Azure the following properties are required."
+    echo "   /p:CloudDropAccountName=\"account name\""
+    echo "   /p:CloudDropAccessToken=\"access token\""
+    echo "   /p:__BuildType=\"Configuration\""
+    echo "Configuration can be Release, Checked, or Debug"
+    exit 1
+}
+
+working_tree_root="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+publish_log=$working_tree_root/publish.log
+
+while [[ $# > 0 ]]
+do
+    unprocessedBuildArgs="$unprocessedBuildArgs $1"
+    shift
+done
+
+# Use uname to determine what the OS is.
+OSName=$(uname -s)
+case $OSName in
+    Linux)
+        __BuildOS=Linux
+        ;;
+
+    Darwin)
+        __BuildOS=OSX
+        ;;
+
+    FreeBSD)
+        __BuildOS=FreeBSD
+        ;;
+
+    OpenBSD)
+        __BuildOS=OpenBSD
+        ;;
+
+    NetBSD)
+        __BuildOS=NetBSD
+        ;;
+
+    SunOS)
+        __BuildOS=SunOS
+        ;;
+
+    *)
+        echo "Unsupported OS $OSName detected, configuring as if for Linux"
+        __BuildOS=Linux
+        ;;
+esac
+
+echo $unprocessedBuildArgs
+
+options="/nologo /v:minimal /flp:v=detailed;Append;LogFile=$publish_log"
+
+echo "Running publish-packages.sh $*" > $publish_log
```

Comment:
$* Is empty at this stage as it was consumed during the loop where you are assigning  `unprocessedBuildArgs